### ISSUE TITLE
ruby pkg needed

### ISSUE THREAD

**Issue by eyberg** [nanovms/ops#73]
this looks like it's loading the ruby interpreter fine - just missing a ton of ruby include shit

```
booting image ...
qemu-system-x86_64 -drive file=image,format=raw,index=0 -display none -serial stdio -nodefaults -no-reboot -m 2G -device isa-debug-exit -drive file=image,format=raw,if=virtio -device virtio-net,netdev=n0 -netdev user,id=n0,hostfwd=tcp::8081-:8081
create fs
kernel complete
pages heap: 000000007fbaf000, length 00000000002f7000
physical memory:
   base 0000000000200000, length 000000007f600000
Using HPET clock source.
init net page alloc 0000000000f027320
read program complete: 00000020000000000 (children:(usr:(children:(lib:(children:(x86_64-linux-gnu:(children:(libgmp.so.10:(filelength:522664 extents:(0:(length:522664 offset:3788800))) libruby-2.3.so.2.3:(filelength:2554168 extents:(0:(length:2554168 offset:1234432))))))))) static:(children:(hw.rb:(filelength:12 extents:(0:(length:12 offset:1233920))))) lib:(children:(x86_64-linux-gnu:(children:(libm.so.6:(filelength:1088952 extents:(0:(length:1088952 offset:4519936))) libc.so.6:(filelength:1868984 extents:(0:(length:1868984 offset:5762560))) libpthread.so.0:(filelength:138696 extents:(0:(length:138696 offset:5623808))) libcrypt.so.1:(filelength:39224 extents:(0:(length:39224 offset:4480512))) libdl.so.2:(filelength:14608 extents:(0:(length:14608 offset:5608960))))))) lib64:(children:(ld-linux-x86-64.so.2:(filelength:162632 extents:(0:(length:162632 offset:4317696))))) ruby2.3:(filelength:6096 extents:(0:(length:6096 offset:4311552))) kernel:(filelength:1230528 extents:(0:(length:1230528 offset:3072)))) environment:() program:/ruby2.3 arguments:(1:hw.rb 0:ruby2.3))
assigned: 10.0.2.15
enq
open /etc/ld.so.cache - not found
open /lib/x86_64-linux-gnu/tls/libruby-2.3.so.2.3 - not found
open /lib/x86_64-linux-gnu/libruby-2.3.so.2.3 - not found
open /usr/lib/x86_64-linux-gnu/tls/libruby-2.3.so.2.3 - not found
open /lib/x86_64-linux-gnu/libgmp.so.10 - not found
nosyscall set_robust_list
open /proc/self/maps - not found
nosyscall getrandom
open /dev/urandom - not found
nosyscall ioctl
nosyscall ioctl
nosyscall ioctl
nosyscall getrandom
open /dev/urandom - not found
nosyscall pipe2
<main>: warning: Failed to create communication pipe for timer thread: No such file or directory
<main>: warning: pipe creation failed for timer: No such file or directory, scheduling broken
nosyscall geteuid
nosyscall getegid
nosyscall getuid
nosyscall geteuid
nosyscall getgid
nosyscall getegid
open /usr/local/lib/site_ruby/2.3.0/enc/encdb.so - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/encdb.so - not found
open /usr/local/lib/site_ruby/enc/encdb.so - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/encdb.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/encdb.so - not found
open /usr/lib/ruby/vendor_ruby/enc/encdb.so - not found
open /usr/lib/ruby/2.3.0/enc/encdb.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/encdb.so - not found
open /usr/local/lib/site_ruby/2.3.0/enc/encdb.so.rb - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/encdb.so.rb - not found
open /usr/local/lib/site_ruby/enc/encdb.so.rb - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/encdb.so.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/encdb.so.rb - not found
open /usr/lib/ruby/vendor_ruby/enc/encdb.so.rb - not found
open /usr/lib/ruby/2.3.0/enc/encdb.so.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/encdb.so.rb - not found
open /usr/local/lib/site_ruby/2.3.0/enc/encdb.so.so - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/encdb.so.so - not found
open /usr/local/lib/site_ruby/enc/encdb.so.so - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/encdb.so.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/encdb.so.so - not found
open /usr/lib/ruby/vendor_ruby/enc/encdb.so.so - not found
open /usr/lib/ruby/2.3.0/enc/encdb.so.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/encdb.so.so - not found
open /usr/local/lib/site_ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/local/lib/site_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/vendor_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/local/lib/site_ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/local/lib/site_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/vendor_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/local/lib/site_ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/ansi_x3_4_1968.so.rb - not found
open /usr/local/lib/site_ruby/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/ruby/vendor_ruby/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/local/lib/site_ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/ansi_x3_4_1968.so.so - not found
open /usr/local/lib/site_ruby/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/ruby/vendor_ruby/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/local/lib/site_ruby/2.3.0/unicode_normalize.rb - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/unicode_normalize.rb - not found
open /usr/local/lib/site_ruby/unicode_normalize.rb - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/unicode_normalize.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/unicode_normalize.rb - not found
open /usr/lib/ruby/vendor_ruby/unicode_normalize.rb - not found
open /usr/lib/ruby/2.3.0/unicode_normalize.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/unicode_normalize.rb - not found
open /usr/local/lib/site_ruby/2.3.0/unicode_normalize.so - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/unicode_normalize.so - not found
open /usr/local/lib/site_ruby/unicode_normalize.so - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/local/lib/site_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/vendor_ruby/enc/ansi_x3_4_1968.so - not found
open /usr/lib/ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/ansi_x3_4_1968.so - not found
open /usr/local/lib/site_ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/ansi_x3_4_1968.so.rb - not found
open /usr/local/lib/site_ruby/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/ruby/vendor_ruby/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/ansi_x3_4_1968.so.rb - not found
open /usr/local/lib/site_ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/enc/ansi_x3_4_1968.so.so - not found
open /usr/local/lib/site_ruby/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/ruby/vendor_ruby/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/enc/ansi_x3_4_1968.so.so - not found
open /usr/local/lib/site_ruby/2.3.0/unicode_normalize.rb - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/unicode_normalize.rb - not found
open /usr/local/lib/site_ruby/unicode_normalize.rb - not found
open /usr/lib/ruby/vendor_ruby/2.3.0/unicode_normalize.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/vendor_ruby/2.3.0/unicode_normalize.rb - not found
open /usr/lib/ruby/vendor_ruby/unicode_normalize.rb - not found
open /usr/lib/ruby/2.3.0/unicode_normalize.rb - not found
open /usr/lib/x86_64-linux-gnu/ruby/2.3.0/unicode_normalize.rb - not found
open /usr/local/lib/site_ruby/2.3.0/unicode_normalize.so - not found
open /usr/local/lib/x86_64-linux-gnu/site_ruby/unicode_normalize.so - not found
open /usr/local/lib/site_ruby/unicode_normalize.so - not found
```

**Comment by mkhon** [comment]
- I no longer see issues with booting fs with large number of files ("create fs" hang)
- nvm (ops) hangs with big manifest are fixed in #150 
- with nanovms/nanos#426 ruby hello world works with the following config:

```
{
    "Files": [ "hello.rb" ],
    "Args": [ "ruby", "hello.rb" ],
    "MapDirs": { "linux-ruby-lib/*": "/usr/lib" }
}
```

linux-ruby-lib was created using the following commands:

```
mkdir linux-ruby-lib
cp -r /usr/lib/ruby linux-ruby-lib
mkdir linux-ruby-lib/x86_64-linux-gnu
cp -r /usr/lib/x86_64-linux-gnu/ruby linux-ruby-lib/x86_64-linux-gnu
rm -rf linux-ruby-lib/ruby/2.3.0/rdoc/generator/template/darkfish # broken symlinks
```

**Comment by eyberg** [comment]
closing this for now - i'm sure there are other bits of crap we need to deal w/but this tkt is good for now

```
➜  r  ops load ruby_2.3.1 -a hello.rb
Extracting /Users/eyberg/.ops/packages/ruby_2.3.1.tar.gz...
[ruby hello.rb]
booting image ...
assigned: 10.0.2.15
yo!
exit_groupexit status 1
➜  r  cat hello.
cat: hello.: No such file or directory
➜  r  cat hello.rb
puts "yo!\n"
```
